### ISSUE TITLE
Python: Scan for requirements in nested directories

### ISSUE THREAD

**Issue by sloria** [dependabot/feedback#113]
I usually put requirements for building docs in `docs/requirements.txt`. Dependabot does not seem to find this file. It'd be great if Dependabot automatically scanned all directories for requirements files (this is the behavior in pyup).

**Comment by sloria** [comment]
This is also a blocker to adopting Dependabot for [cookiecutter-flask](https://github.com/sloria/cookiecutter-flask), which has its requirements nested within the `{{cookiecutter.app_name}}` directory.

**Comment by greysteil** [comment]
Thanks for the feedback - super helpful.

Dependabot doesn't ever clone your repo, so scanning all directories for requirements files is a little tricky for us. Currently we pick up any files with `requirements` in their name in whatever directory you're specified to Dependabot (by default the root), and any files in any 1st-level subdirectories named `requirements`. You can see to code [here](https://github.com/dependabot/dependabot-core/blob/master/lib/dependabot/file_fetchers/python/pip.rb#L85-L97).

Can I run a couple of options for improvement passed you?
1) It's already possible to specify a directory for Dependabot to run against when adding a project to Dependabot. You can add several update configs, each with a different directory to the same project, so it would be possible for you to add one with the root directory and one with the `docs` directory. That would require no change on Dependabot's side, but feels a bit janky.
2) I could change Dependabot's behaviour so that it fetches all files that end with `.txt` in the root directory or any 1st-level subdirectories. I'm pretty sure that would fix the above, and work for 99% of people.
3) Something else I haven't thought of?

**Comment by sloria** [comment]
Thanks for the quick reply.

> 2. I could change Dependabot's behaviour so that it fetches all files that end with .txt in the root directory or any 1st-level subdirectories.

That would certainly address most use cases, including the two I posted about. 👍

**Comment by greysteil** [comment]
Cool, that's what I figured. I'll get it done!

**Comment by greysteil** [comment]
OK, I've just deployed that change. However, there's still a check that there's some kind of requirements file, or a folder called `requirements` at the level at which you add Dependabot . For `cookiecutter-flask` that means you'd need to set the directory (in the advanced options when adding a repo / language) as `{{cookiecutter.app_name}}` - I've tried it on a fork and with that set everything works great.

Since the dev requirements for `cookiecutter-flask` as nested at `{{cookiecutter.app_name}}/requirements/dev.txt` Dependabot will also pick them up if you set the directory as `{{cookiecutter.app_name}}`.

Hopefully the requirement of having some requirements file/folder in the directory you specify Dependabot to operate on is fine?

**Comment by sloria** [comment]
Actually, I am evaluating Dependabot for use with one of my work projects, [osf.io](https://github.com/CenterForOpenScience/osf.io). It has a number of "addons", which are pluggable Django apps which have their requirements in their own directory, which is two deep, e.g. [`addons/dropbox/requirements.txt`](https://github.com/CenterForOpenScience/osf.io/blob/develop/addons/dropbox/requirements.txt). Is there any way to support this?

**Comment by greysteil** [comment]
Looks like a good solution there would be to be able to install Dependabot on the `addons` directory and have it pick up all of the requirements.txt files one level deep. Would that work?

I'll have to remove the requirement of having some kind of requirement file/folder at the top level, but I'm happy to do that if it'll make a difference to you.

**Comment by sloria** [comment]
Yeah, I would need a way for Dependabot to scan two deep, `/addons/*/requirements.txt` and at the top level, `/requirements/*.txt`.

**Comment by greysteil** [comment]
Would you be happy adding Dependabot twice for Python for that repo? Once at the root level and one at the `addons/` level? It will mean you get PRs separately for updates to the files in `/requirements/*.txt` and those in `/addons/*/requiments.txt`, but hopefully that's fine?

For example, here's the setup we have for keeping Dependabot core's JS requirements up-to-date in two different repos:

![image](https://user-images.githubusercontent.com/1144873/43404984-b9941f4a-9410-11e8-8ed7-281a8d1b37c9.png)


(You could already add Dependabot many times over, once for each addon folder, to that repo, but that sounds pretty annoying.)

**Comment by sloria** [comment]
Oh, I didn't realize you could add a language twice for the same repo. Yeah, that will probably be fine for our use case. I'll give it a shot when I get time to do this in earnest.

Thanks!
